src/api/v4/dataview/category.js

```diff
@@ -0,0 +1,185 @@
+var _ = require('underscore');
+var Base = require('./base');
+var constants = require('../constants');
+var CategoryDataviewModel = require('../../../dataviews/category-dataview-model');
+var CategoryFilter = require('../../../windshaft/filters/category');
+
+/**
+ * Category dataview object
+ *
+ * @param {carto.source.Base} source - The source where the datavew will fetch the data.
+ * @param {string} column - The column name to get the data.
+ * @param {object} options
+ * @param {carto.operation} options.operation - The operation to apply to the data.
+ * @param {string} options.operationColumn - The column name used in the operation.
+ *
+ * @constructor
+ * @extends carto.dataview.Base
+ * @memberof carto.dataview
+ * @api
+ */
+function Category (source, column, options) {
+  this._initialize(source, column, options);
+}
+
+Category.prototype = Object.create(Base.prototype);
+
+/**
+ * Set the dataview operation
+ *
+ * @param  {carto.operation} operation
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setOperation = function (operation) {
+  this._checkOperation(operation);
+  this._options.operation = operation;
+  if (this._internalModel) {
+    this._internalModel.set('aggregation', operation);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview operation
+ *
+ * @return {carto.operation} Current dataview operation
+ * @api
+ */
+Category.prototype.getOperation = function () {
+  return this._options.operation;
+};
+
+/**
+ * Set the dataview operationColumn
+ *
+ * @param  {string} operationColumn
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setOperationColumn = function (operationColumn) {
+  this._checkOperationColumn(operationColumn);
+  this._options.operationColumn = operationColumn;
+  if (this._internalModel) {
+    this._internalModel.set('aggregation_column', operationColumn);
+  }
+  return this;
+};
+
```

Comment:
We've talked about grouping `setOperation` and `setOperationColumn` for providing the user the option of setting them at once.

Reply:
I go with `setOperation(operation)`, `setOperation(operation, column)` and `setOperationColumn(column)`

Reply:
I talk to solutions and I think it is better to go with individual set/get methods for all the dataviews and then wait for the feedback of the internal release to add new group functions.